api/catalog/api/models/mixins.py

```diff
@@ -27,6 +27,10 @@ class ForeignIdentifierMixin(models.Model):
     """
     This mixin adds fields related to the external unique ID to any model. Do
     not use this as the sole base class.
+
+    This mixin adds
+
+    - foreign_identifier: CharField
```

Comment:
Yes, this is helpful when hovering you're reading a class and hover over the name of it's mixins to see what fields they're adding to the model.

Reply:
Gotcha. My concern is that because it is essentially a 1:1 mirror of the implementation that it will easily go out of date quickly (as it already had, with this line needing to be move from one serializer to another without any changes on the actual implementation of either one).

If our base mixin classes are so abstract that it's not easy to remember (or learn) what fields they provide based on the names and reading the declarative code inside their definitions, maybe it's a sign that things are too complex?

It could be I'm over thinking this though. I'm just wary of whenever we start to write documentation that says in literal terms what the code immediately below it says.

Reply:
I don't like this either, and I'm 100% in agreement with you that the mixins are super-confusing and can be improved. If we did spend a lot of time in the API models (which we don't), we'd soon remember what field was where.

But at the moment, it's a bit hard to see a class inheriting from 4-5 mixins and having to guess which fields are coming from which mixin (or going to each of their definitions to check).